drivers/npmx/npmx_shell.c

```diff
@@ -3414,164 +3422,157 @@ static npmx_timer_mode_t timer_mode_int_convert_to_enum(uint32_t timer_mode)
+	case TIMER_CONFIG_PARAM_MODE:
+		timer_config.mode = timer_mode_int_convert_to_enum(shell, config_value);
 		if (timer_config.mode == NPMX_TIMER_MODE_INVALID) {
-			shell_error(shell,
-				    "Error: timer mode can be: 0-boot monitor, 1-watchdog warning, "
-				    "2-watchdog reset, 3-general purpose or 4-wakeup.");
 			return 0;
 		}
-	} else if (config_type == TIMER_CONFIG_TYPE_PRESCALER) {
-		if (input_value < NPMX_TIMER_PRESCALER_COUNT) {
-			timer_config.prescaler = (input_value ? NPMX_TIMER_PRESCALER_FAST :
-								NPMX_TIMER_PRESCALER_SLOW);
-		} else {
-			shell_error(shell, "Error: timer prescaler value can be 0-slow or 1-fast.");
+		break;
+	case TIMER_CONFIG_PARAM_PRESCALER:
+		if (config_value >= NPMX_TIMER_PRESCALER_COUNT) {
+			shell_error(shell, "Error: Wrong prescaler:");
+			print_hint_error(shell, 0, "Slow");
+			print_hint_error(shell, 1, "Fast");
 			return 0;
 		}
-	} else if (config_type == TIMER_CONFIG_TYPE_COMPARE) {
-		if (input_value <= NPM_TIMER_COUNTER_COMPARE_VALUE_MAX) {
-			timer_config.compare_value = input_value;
-		} else {
-			shell_error(shell, "Error: timer period value can be 0..0x%lX.",
+
+		timer_config.prescaler =
+			(config_value ? NPMX_TIMER_PRESCALER_FAST : NPMX_TIMER_PRESCALER_SLOW);
+		break;
+	case TIMER_CONFIG_PARAM_COMPARE:
+		if (config_value >= NPM_TIMER_COUNTER_COMPARE_VALUE_MAX) {
+			shell_error(shell, "Error: wrong period: value can be 0..0x%lX.",
 				    NPM_TIMER_COUNTER_COMPARE_VALUE_MAX);
 			return 0;
 		}
+
+		timer_config.compare_value = config_value;
+		break;
 	}
 
 	err_code = npmx_timer_config_set(timer_instance, &timer_config);
+	if (!check_error_code(shell, err_code)) {
+		print_set_error(shell, "timer config");
+		return 0;
+	}
 
-	if (check_error_code(shell, err_code)) {
-		if (config_type == TIMER_CONFIG_TYPE_MODE) {
-			shell_print(shell, "Success: %d.", timer_config.mode);
-		} else if (config_type == TIMER_CONFIG_TYPE_PRESCALER) {
-			shell_print(shell, "Success: %d.", timer_config.prescaler);
-		} else if (config_type == TIMER_CONFIG_TYPE_COMPARE) {
-			shell_print(shell, "Success: %d.", timer_config.compare_value);
-		}
-	} else {
-		shell_error(shell, "Error: unable to set timer configuration.");
+	print_success(shell, config_value, UNIT_TYPE_NONE);
+	return 0;
+}
+
+static int timer_config_get(const struct shell *shell, timer_config_param_t config_type)
+{
+	npmx_timer_t *timer_instance = timer_instance_get(shell);
+	if (timer_instance == NULL) {
+		return 0;
+	}
+
+	npmx_timer_config_t timer_config;
+	npmx_error_t err_code = npmx_timer_config_get(timer_instance, &timer_config);
+	if (!check_error_code(shell, err_code)) {
+		print_get_error(shell, "timer config");
+		return 0;
 	}
 
+	switch (config_type) {
+	case TIMER_CONFIG_PARAM_MODE:
+		print_value(shell, (int)timer_config.mode, UNIT_TYPE_NONE);
+		break;
+	case TIMER_CONFIG_PARAM_PRESCALER:
+		print_value(shell, (int)timer_config.prescaler, UNIT_TYPE_NONE);
+		break;
+	case TIMER_CONFIG_PARAM_COMPARE:
```

Comment:
I have noticed you try to keep alphabetical order also in switch cases, so according to that, `compare` should be first.